### PR TITLE
cmake updates

### PR DESCRIPTION
**Pull request by pca006132**
There are three targets now: manifold, manifoldc, meshIO (when MANIFOLD_CBIND/MANIFOLD_EXPORT are ON). Verified that we can install libmanifold and use cmake now:

```cmake
cmake_minimum_required(VERSION 3.18)
project(testing LANGUAGES CXX)
set(CMAKE_CXX_STANDARD 17)
set(CMAKE_CXX_STANDARD_REQUIRED True)
find_package(manifold "2.3.1" REQUIRED)
add_executable(testing test.cpp)
target_link_libraries(testing PRIVATE manifold)
```

```cpp
#include "manifold.h"
int main() {
    manifold::Manifold foo;
    return 0;
}
```

Also factored out clipper2 so users can use system clipper2 instead of our submodule.
Currently there are no breaking changes.

Remaining things (breaking changes):
1. We should probably rename our header files to use less generic names, as we are polluting the global namespace now (they are dump into `/usr/local/include`...).
2. `MANIFOLD_PAR` change to boolean.

### CODE REVIEW

**Comment by pca006132** [comment]
also testing the python builds: https://github.com/pca006132/manifold/actions/runs/7506118399

**Comment by pca006132** [comment]
I should probably add a cmake user test in the CI as well, will do tmr.

**Comment by pca006132** [comment]
@ramcdona @t-paul you seems to be interested and experienced with cmake, what do you think about the changes here?

**Comment by pca006132** [comment]
For some reason `sudo cmake --install .` failed on macos CI. Not sure why the privilege is not enough.

**Comment by pca006132** [comment]
@elalish what do you think about changing the header names? Should we do this here or another PR? This may break some user's code, not sure if there are users directly using `public.h` for example.

**Comment by elalish** [comment]
I'm not quite sure of the pros/cons. Is there some kind of standard practice for C++ libraries to name their headers? Seems like they must have some system to avoid name collisions, right?

**Comment by pca006132** [comment]
The standard way is to put them under a directory, but this requires changing our directory structure as well.

**Comment by elalish** [comment]
Perhaps we should consider that - what would a more standard directory structure look like? This could be part of v3.0.

**Comment by pca006132** [comment]
I think we can leave it for later, so this PR is ready for review and merging.

To be honest I have no idea what a "typical layout" looks like when there are several simple modules...

File: flake.nix [hunk 1]
```diff
@@ -5,44 +5,25 @@
...
-            version = "24486a1";
-            src = builtins.fetchGit {
-              url = "https://github.com/NVIDIA/thrust.git";
-              ref = "main";
-              rev = "24486a169a62a58ef8f824d3dc9613c006b6f5a7";
-              submodules = true;
-            };
-            cmakeFlags = [ "-DTHRUST_ENABLE_HEADER_TESTING=OFF" "-DTHRUST_ENABLE_TESTING=OFF" "-DTHRUST_ENABLE_EXAMPLES=OFF" "-DTHRUST_DEVICE_SYSTEM=CPP" ];
-            fixupPhase = ''
-              cat <<EOT > $out/lib/cmake/thrust/thrust-header-search.cmake
```
Comment (elalish) [inline flake.nix:26]: Nice to remove all this, thanks!

**Review by elalish — APPROVED** [review]
Dang, this look excellent; thanks for all the cleanup! I'll let you merge it, in case there's anything you want to add. LGTM.

File: CMakeLists.txt [hunk 2]
```diff
@@ -108,3 +122,29 @@ if(MANIFOLD_TEST)
   add_subdirectory(test)
   add_subdirectory(extras)
 endif()
+
+# installation related
+
+set(MANIFOLD_VERSION_MAJOR 2)
+set(MANIFOLD_VERSION_MINOR 3)
+set(MANIFOLD_VERSION_PATCH 1)
```
[the remark this answers is not in the document]
Comment (pca006132) [reply CMakeLists.txt:130]: yes, will add

**Review by elalish — APPROVED** [review]

**Pull request merged**